plugins/frontend/backstage-plugin-jira/src/api/index.ts

```diff
@@ -241,66 +190,39 @@ export class JiraAPI {
     }
     const project = (await request.json()) as Project;
 
-    // If component not defined, execute the same code. Otherwise use paged request
-    // to fetch also the issue-keys of all the tasks for that component.
-    let issuesCounter: IssuesCounter[] = [];
-    let ticketIds: string[] = [];
-    let tickets: Ticket[] = [];
-    const foundIssues = await this.getIssueCountPaged({
-      apiUrl,
-      projectKey,
-      component,
-      label,
-      statusesNames,
-    });
-    if (!component && !label) {
-      // Generate counters for each issue type
-      const issuesTypes = project.issueTypes.map((status: IssueType) => ({
-        name: status.name,
-        iconUrl: status.iconUrl,
-      }));
-
-      const filteredIssues = issuesTypes.filter(el => el.name !== 'Sub-task');
-
-      issuesCounter = await Promise.all(
-        filteredIssues.map(issue => {
-          const issueType = issue.name;
-          const issueIcon = issue.iconUrl;
-          return this.getIssuesCountByType({
-            apiUrl,
-            projectKey,
-            component,
-            statusesNames,
-            issueType,
-            issueIcon,
-            label,
-          });
-        }),
-      );
-    } else {
-      // Get all issues, count them using reduce and generate a ticketIds array,
-      // used to filter in the activity stream
-      const issuesTypes = project.issueTypes.map(
+    const issuesTypes = project.issueTypes
+      .filter(issueType => issueType.name !== 'Sub-task')
+      .map(
         (status: IssueType): IssuesCounter => ({
           name: status.name,
           iconUrl: status.iconUrl,
           total: 0,
         }),
       );
-      issuesCounter = foundIssues
-        .reduce((prev, curr) => {
-          const name = curr.fields?.issuetype.name;
-          const idx = issuesTypes.findIndex(i => i.name === name);
-          if (idx !== -1) {
-            issuesTypes[idx].total++;
-          }
-          return prev;
-        }, issuesTypes)
-        .filter(el => el.name !== 'Sub-task');
-
-      ticketIds = foundIssues.map(i => i.key);
-    }
-    tickets = foundIssues.map(index => {
+
+    const foundIssues = await this.getIssuesPaged({
+      apiUrl,
+      projectKey,
+      component,
+      label,
+      statusesNames,
+    });
+
+    const issuesCounter: IssuesCounter[] = issuesTypes.map(issueType => {
```

Comment:
You might be better off using `lodash` `groupBy` or reducing it manually in here to avoid having the need to loop through the issues multiple times. I don't know how big the `issueTypes` collection grows, so it could be fine on smaller amounts.

Reply:
Good point, I decided to just "merge" the issueTypes and issueCounter loops, as there was really no reason do have them separate.